### PR TITLE
Raise handled exceptions

### PR DESCRIPTION
**Pull request by aljazkosir**
Addresses issue https://github.com/xlab-si/redfish-client-python/issues/8

\cc @miha-plesko @tadeboro

### CODE REVIEW

File: redfish_client/connector.py [hunk 1]
```diff
@@ -51,7 +51,11 @@ def _url(self, path):
 
     def _request(self, method, path, payload=None):
         args = dict(json=payload) if payload else {}
-        resp = self._client.request(method, self._url(path), **args)
+        try:
+            resp = self._client.request(method, self._url(path), **args)
+        except requests.exceptions.ConnectionError:
+            raise InaccessibleException(
+                'Endpoint at {} is not accessible'.format(self._base_url))
```
Comment (tadeboro) [inline redfish_client/connector.py]: Nitpick: We should use `"` in place of `'` for consistency with the rest of the code base.

**Comment by aljazkosir** [comment]
@tadeboro I added unit tests for newly implemented exceptions and I also added unit tests for other exceptions that were missing them (I included them in this PR since we are refactoring exceptions, but I can also create new PR if wish)

**Comment by tadeboro** [comment]
Sorry for the noise that I caused by closing and reopening this pull request. I misconfigured the CircleCI when I set up the integration. Closing and reopening triggered the build.

**Comment by tadeboro** [comment]
Thank you for your contribution @aljazkosir.

**Pull request merged**